src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+                    args.Dispose();
+                    if (response == null)
+                        throw new Exception("Internal error marshalling the response Promise from `fetch`.");
+
+                    JSObject t = (JSObject)await response.ConfigureAwait(continueOnCapturedContext: true);
+
+                    var status = new WasmFetchResponse(t, abortController, abortCts, abortRegistration);
+
+                    HttpResponseMessage httpresponse = new HttpResponseMessage((HttpStatusCode)Enum.Parse(typeof(HttpStatusCode), status.Status.ToString()));
+
+                    bool streamingEnabled = request.Properties.TryGetValue("WebAssemblyEnableStreamingResponse", out object? streamingEnabledValue) && (bool)(streamingEnabledValue ?? false);
+
+                    httpresponse.Content = StreamingSupported && streamingEnabled
+                        ? new StreamContent(wasmHttpReadStream = new WasmHttpReadStream(status))
+                        : (HttpContent)new BrowserHttpContent(status);
+
+                    // Fill the response headers
+                    // CORS will only allow access to certain headers.
+                    // If a request is made for a resource on another origin which returns the CORs headers, then the type is cors.
+                    // cors and basic responses are almost identical except that a cors response restricts the headers you can view to
+                    // `Cache-Control`, `Content-Language`, `Content-Type`, `Expires`, `Last-Modified`, and `Pragma`.
+                    // View more information https://developers.google.com/web/updates/2015/03/introduction-to-fetch#response_types
+                    //
+                    // Note: Some of the headers may not even be valid header types in .NET thus we use TryAddWithoutValidation
+                    using (JSObject respHeaders = status.Headers)
+                    {
+                        if (respHeaders != null)
+                        {
+                            using (var entriesIterator = (JSObject)respHeaders.Invoke("entries"))
+                            {
+                                JSObject? nextResult = null;
+                                try
+                                {
+                                    nextResult = (JSObject)entriesIterator.Invoke("next");
+                                    while (!(bool)nextResult.GetObjectProperty("done"))
+                                    {
+                                        using (var resultValue = (Interop.JavaScript.Array)nextResult.GetObjectProperty("value"))
+                                        {
+                                            var name = (string)resultValue[0];
+                                            var value = (string)resultValue[1];
+                                            if (!httpresponse.Headers.TryAddWithoutValidation(name, value))
+                                                if (httpresponse.Content != null)
```

Comment:
The code above always assigns this, so this will never be null, right? (We should ensure it's not.)

Reply:
Yes you are right.  The null check was left over from older code.  Addressed in recent commit